### PR TITLE
Store instance of Destination in DestinationConfig

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/microsoft/reverse-proxy/issues/788

### CODE REVIEW

**Comment by Tratcher** [comment]
This is consistent with ClusterConfig. The open question is if DestinationConfig serves any purpose or if it can be eliminated? It does not track any additional state. I'll bring this up in our Tuesday meeting and get back to you.

**Comment by Kahbazi** [comment]
> This is consistent with ClusterConfig. The open question is if DestinationConfig serves any purpose or if it can be eliminated? It does not track any additional state. I'll bring this up in our Tuesday meeting and get back to you.

Maybe it could be used for custom load balancing policies that need state per destination.

**Comment by Tratcher** [comment]
> > This is consistent with ClusterConfig. The open question is if DestinationConfig serves any purpose or if it can be eliminated? It does not track any additional state. I'll bring this up in our Tuesday meeting and get back to you.
> 
> Maybe it could be used for custom load balancing policies that need state per destination.

When we looked at doing something similar for passive health checks we found that using ConditionalWeakTable to store the data locally worked better.
https://github.com/microsoft/reverse-proxy/blob/1fb29bbe27ff316f06996bc08eabc17cb0e79282/src/ReverseProxy/Service/HealthChecks/TransportFailureRateHealthPolicy.cs#L34

File: src/ReverseProxy/Service/RuntimeModel/DestinationConfig.cs [hunk 1]
```diff
@@ -37,5 +42,10 @@ public DestinationConfig(string address, string health)
         /// Endpoint accepting active health check probes.
         /// </summary>
         public string Health { get; }
```
Comment (Tratcher) [inline src/ReverseProxy/Service/RuntimeModel/DestinationConfig.cs:44]: Why keep these properties if they're duplicates from the Destination?

Comment (Kahbazi) [reply src/ReverseProxy/Service/RuntimeModel/DestinationConfig.cs:44]: I will remove them.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
We discussed this briefly but didn't come to any conclusions. We'll revisit it when we do a fine grained API review.

**Pull request merged**